[TwigBundle] Reconfigure twig paths when they are updated

chbruyand opened the pull request:
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14771, #14768, #14262, #14778 |
| License | MIT |

Refresh twig paths upon creation and deletion. As we don't care neither about path's modification time nor path's content, a new Resource has been added in the Config Component.
Full discussion in #14778.

chbruyand committed:
[TwigBundle] Reconfigure twig paths when they are updated

chbruyand committed:
fix wrong copy/paste

mpdude commented:
Not sure, but does `src/Symfony/Bundle/TwigBundle/composer.json` needs an update for `symfony/config`?

chbruyand commented:
I think so, but don't know which version to put there

mpdude commented:
I'd go for ~2.8, which is the first version that will contain the new resource.

chbruyand committed:
Next version of symfony/config is needed

sstok commented on src/Symfony/Component/Config/Resource/FileExistenceResource.php:
```diff
@@ -0,0 +1,78 @@
+     */
+    public function __toString()
+    {
+        return (string) $this->resource;
```
You can do the string casting in the constructor already.

aitboudad commented:
:+1:

chbruyand committed:
Update FileExistenceResource.php

chbruyand commented:
Any other suggestion / improvement ?

stof commented on src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php:
```diff
@@ -74,23 +75,30 @@ public function load(array $configs, ContainerBuilder $container)
             } else {
                 $twigFilesystemLoaderDefinition->addMethodCall('addPath', array($path, $namespace));
             }
+            $container->addResource(new FileExistenceResource($path));
```
not needed here as we don't change the container configuration based on the existence of this folder

chbruyand replied:
Thanks for the hint

chbruyand committed:
Remove unnecessary tracked path

https://github.com/symfony/symfony/pull/14781/files#r31505368

not needed here as we don't change the container configuration based on
the existence of this folder

stof commented:
:+1:

fabpot commented:
Thank you @chbruyand.
